docs/source/cpp/compute.rst

```diff
@@ -1475,16 +1480,13 @@ value, but smaller than nulls.
 * \(2) The output is an array of indices into the input, that define a
   stable sort of the input.
 
-* \(3) Input values are ordered lexicographically as bytestrings (even
-  for String arrays).
-
-* \(4) The input must be an array. The default order is ascending.
+* \(3) The input must be an array. The default order is ascending.
```

Comment:
Consider adding here: "By default, nulls and NaNs are sorted after any non-null values."

Reply:
It's already above in the section summary.